### ISSUE TITLE
D3D11 profiling problems in 0.10

### ISSUE THREAD

**Issue by niello** [wolfpld/tracy#670]
1. `is_active` var is referenced but there is only `active`. Compilation error. https://github.com/wolfpld/tracy/blob/116e82de9ebbf3937b066a3be23b8d8de16b2a0a/public/tracy/TracyD3D11.hpp#L360
2. In a regular run an application never reaches here, `tcpu` and `tgpu` remain 0 and GPU track is not drawn. But manually walking this part under debugger seems to give GPU enough time to finish the query and then data appear. https://github.com/wolfpld/tracy/blob/116e82de9ebbf3937b066a3be23b8d8de16b2a0a/public/tracy/TracyD3D11.hpp#L114-L115
3. Pressing "Auto" button in this situation crashes the client
![image](https://github.com/wolfpld/tracy/assets/2044092/bb02350d-3f19-4dc1-a82e-a0d661d3fbf4)
4. On init D3D11 spawns a warning: D3D11 WARNING: ID3D11DeviceContext::End: End is being invoked on a Query, where the previous results have not been obtained with GetData. This is valid; but unusual. The previous results are being abandoned, and new Query results will be generated. [ EXECUTION WARNING 410: QUERY_END_ABANDONING_PREVIOUS_RESULTS]

Release 0.10
Windows, D3D11
On-demand connection mode

**Comment by wolfpld** [comment]
@slomp Can you take a look?

**Comment by slomp** [comment]
Sure, I'll try to investigate it this week!

**Comment by simonvanbernem** [comment]
I also ran into these issues, with 0.10.0. Are they being worked on. `is_active` was already fixed in https://github.com/wolfpld/tracy/issues/753, but I'm also seing the d3d11 warnings being generated.

**Comment by slomp** [comment]
> but I'm also seeing the d3d11 warnings being generated.

Which warnings? Only this one?
```
warning: D3D11 WARNING: ID3D11DeviceContext::End: End is being invoked on a Query, where the previous results have not been obtained with GetData. This is valid; but unusual. The previous results are being abandoned, and new Query results will be generated. [ EXECUTION WARNING 410: QUERY_END_ABANDONING_PREVIOUS_RESULTS]
```

**Comment by simonvanbernem** [comment]
> > but I'm also seeing the d3d11 warnings being generated.
> 
> Which warnings? Only this one?
> 
> ```
> warning: D3D11 WARNING: ID3D11DeviceContext::End: End is being invoked on a Query, where the previous results have not been obtained with GetData. This is valid; but unusual. The previous results are being abandoned, and new Query results will be generated. [ EXECUTION WARNING 410: QUERY_END_ABANDONING_PREVIOUS_RESULTS]
> ```

yes

**Comment by simonvanbernem** [comment]
Although I also get a crash in TracyD3D11.hpp when connected to the server and vsync turned off in a scene where my game runs at >1000fps:

Assertion failed: false && "TracyD3D11: " "timestamp expected to be ready, but it was not!", file C:\Workspaces\spaced\source\tracy\public/tracy/TracyD3D11.hpp, line 224

Should I open a new issue for that, or is there some information I could give that would be helpful?

**Comment by slomp** [comment]
> when connected to the server and vsync turned off in a scene where my game runs at >1000fps:

Are you calling `TracyD3D11Collect` periodically when vsync off?

**Comment by simonvanbernem** [comment]
> > when connected to the server and vsync turned off in a scene where my game runs at >1000fps:
> 
> Are you calling `TracyD3D11Collect` periodically when vsync off?

Yes, immediatedly after IDXGISwapChain_Present. I have three calls to ID3D11DeviceContext_End in my own code that I use(d) to measure the GPU frame times before I used tracy. Might that interfere?

**Comment by slomp** [comment]
Would you by any chance have a small, dependency-free minimal reproducible code that you could share?
(I dug some old D3D11 code I had and tried to reproduce the issue, but was unsuccessful).

**Comment by simonvanbernem** [comment]
> Would you by any chance have a small, dependency-free minimal reproducible code that you could share?
> (I dug some old D3D11 code I had and tried to reproduce the issue, but was unsuccessful).

No, sorry. Only a big application that doesn't even repro it 100% of the time.

**Comment by slomp** [comment]
PR open to address the `is_active` compilation error with `TRACY_ON_DEMAND`:
https://github.com/wolfpld/tracy/pull/806

**Comment by simonvanbernem** [comment]
Hey @slomp, I managed to make a self contained repro for the warning. Please take a look at #947

**Comment by wolfpld** [comment]
4db02dd041be
